spring-boot/config-server/src/main/java/eu/coatrack/admin/server/config/DataSourceConfiguration.java

```diff
@@ -0,0 +1,77 @@
+package eu.coatrack.admin.server.config;
+
+/*-
+ * #%L
+ * coatrack-config-server
+ * %%
+ * Copyright (C) 2013 - 2021 Corizon | Institut für angewandte Systemtechnik Bremen GmbH (ATB)
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.jdbc.DataSourceProperties;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Primary;
+
+import javax.sql.DataSource;
+
+@Configuration
+public class DataSourceConfiguration {
+
+    @Primary
+    @Bean
+    public DataSource getDataSource(
+            @Qualifier("first") DataSourceProperties first,
```

Comment:
@gr-hovest-atb double-check how the previous approach worked when starting the config server locally without postgres

Reply:
I checked it again without running postgresql server, the previous version actually then throws several exceptions on startup, complaining that postgresql is not "listening" on its port.

My suggestion would be: 
- create a new file `application-postgresql.yml` 
- move the postgresql-specific config into that file, so that it looks roughly like this:
```
spring:
   datasource:
      username: postgres
      password: postgres1234
      url: jdbc:postgresql://localhost/ygg_config_server
      driver: org.postgresql.Driver
    jpa:
        properties:
            hibernate:
                dialect: org.hibernate.dialect.PostgreSQLDialect
```
Then a local dev startup should automatically start-up an in-memory h2, while in production use the profile `postgresql` should be used.

Reply:
Done as suggested.

Reply:
TODO
- re-add the `jpa`
- remove duplicate stuff from application-postgres.yml

Reply:
Done as suggested.